src/loaders/lottie/tvgLottieProperty.h

```diff
@@ -432,46 +432,75 @@ struct LottiePathSet : LottieProperty
-            else {
-                _copy(path, cmds);
-                _copy(path, pts, transform);
+            if (roundness) {
+                if (!offsetPath) {
+                    roundness->appendPath(path->cmds, path->cmdsCnt, path->pts, path->ptsCnt, cmds, pts, transform);
+                    return true;
+                }
+                Array<PathCommand> cmds1(path->cmdsCnt);
+                Array<Point> pts1(path->ptsCnt);
+                roundness->appendPath(path->cmds, path->cmdsCnt, path->pts, path->ptsCnt, cmds1, pts1, transform);
+                offsetPath->append(cmds1, pts1, cmds, pts, true);
+                return true;
+            }
+            if (offsetPath) {
+                offsetPath->appendPath(path->cmds, path->cmdsCnt, path->pts, path->ptsCnt, cmds, pts, true);
+                return true;
             }
+            _copy(path, cmds);
+            _copy(path, pts, transform);
             return true;
         }
 
         auto s = frame->value.pts;
         auto e = (frame + 1)->value.pts;
 
-        if (roundness) {
-            auto interpPts = (Point*)malloc(frame->value.ptsCnt * sizeof(Point));
-            auto p = interpPts;
-            for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e, ++p) {
-                *p = lerp(*s, *e, t);
-                if (transform) *p *= *transform;
-            }
-            roundness->appendPath(frame->value.cmds, frame->value.cmdsCnt, interpPts, frame->value.ptsCnt, cmds, pts, nullptr);
-            free(interpPts);
-            return true;
-        } else {
+        if (!roundness && !offsetPath) {
             for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e) {
                 auto pt = lerp(*s, *e, t);
                 if (transform) pt *= *transform;
                 pts.push(pt);
             }
             _copy(&frame->value, cmds);
+            return true;
         }
+
+        auto interpPts = (Point*)malloc(frame->value.ptsCnt * sizeof(Point));
+        auto p = interpPts;
+        for (auto i = 0; i < frame->value.ptsCnt; ++i, ++s, ++e, ++p) {
+            *p = lerp(*s, *e, t);
+            if (transform) *p *= *transform;
+        }
+
+        if (roundness) {
+            if (!offsetPath) roundness->appendPath(frame->value.cmds, frame->value.cmdsCnt, interpPts, frame->value.ptsCnt, cmds, pts, nullptr);
+            else {
+                Array<PathCommand> cmds1;
+                Array<Point> pts1;
+                roundness->appendPath(frame->value.cmds, frame->value.cmdsCnt, interpPts, frame->value.ptsCnt, cmds1, pts1, nullptr);
+                offsetPath->append(cmds1, pts1, cmds, pts, true);
+            }
+        } else if (offsetPath) {
+            offsetPath->appendPath(frame->value.cmds, frame->value.cmdsCnt, interpPts, frame->value.ptsCnt, cmds, pts, true);
+        }
+
+        free(interpPts);
+
+        if (roundness) {
```

Comment:
one line.

Reply:
this is handled in l.476, some bug after rebasing